### PR TITLE
Refactor New moon

### PR DESCRIPTION
**Pull request by neihousaigaai**
Create `new_moon` folder:

- move NewMoonMode class into `new_moon/__init__.py`
- create NewMoonEvent base class
- create each event class by inherited NewMoonEvent class in 1 file
- check active in event + phase due to Punishment event action

Also show new moon event name in status table:

image

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -63,8 +63,9 @@ async def test_case(game, filepath):
         timeline_action_list = test_case_data["timeline"]
         for timeline_idx, action_data in enumerate(timeline_action_list):
             await asyncio.sleep(delay_time)
-
             assert check_alive_players(game, list(map(lambda x: id_map[x], action_data["alive"])), playersname)
+            await asyncio.sleep(delay_time)
+
```
Comment (nitro2) [inline test.py:68]: Why do we need this change?

File: game/__init__.py [hunk 2]
```diff
@@ -879,13 +872,13 @@ async def do_end_daytime_phase(self):
             print("lynched list:", self.voter_dict)
             self.voter_dict = {}
 
-        if self.modes.get("new_moon", False) and self.new_moon_mode.current_event == NewMoonMode.HEADS_OR_TAILS:
-            coin_toss_value = self.new_moon_mode.do_coin_toss()
+        if self.new_moon_mode.get_current_event() is HeadsOrTails:
+            coin_toss_value = await self.new_moon_mode.do_action(self.interface)
             print("coin toss value =", coin_toss_value)
```
Comment (nitro2) [inline game/__init__.py:877]: Duplicated print inside the event

File: game/__init__.py [hunk 3]
```diff
@@ -879,13 +872,13 @@ async def do_end_daytime_phase(self):
             print("lynched list:", self.voter_dict)
             self.voter_dict = {}
 
-        if self.modes.get("new_moon", False) and self.new_moon_mode.current_event == NewMoonMode.HEADS_OR_TAILS:
-            coin_toss_value = self.new_moon_mode.do_coin_toss()
+        if self.new_moon_mode.get_current_event() is HeadsOrTails:
+            coin_toss_value = await self.new_moon_mode.do_action(self.interface)
```
Comment (nitro2) [inline game/__init__.py:876]: I don't think we need both `self.new_moon_mode.do_action(self.interface)` and `self.new_moon_mode.do_end_daytime_phase(self.interface, coin_toss_value=coin_toss_value)`
Just need 1 `coin_toss_value = self.new_moon_mode.do_end_daytime_phase(self.interface)` here.

File: test.py (hunk 1 above)
Comment (neihousaigaai) [reply test.py:68]: It prevents randomly failing `case-pathologist-autopsy-super-and-lycan-and-tanner-werewolf-win` test case.

**Review by nitro2 — APPROVED** [review]

**Pull request merged**